### PR TITLE
Display hide navbar

### PR DESCRIPTION
**Pull request by BenSolt**
# Description
This fix properly routes the proper navigation component to it's respective pages, earlier we had the authenticated navigation showing on login and register and visa versa for the register navigation. This branch also adds testing to the app component to ensure the proper component is displayed. Additionally, code-cleanup was performed. 

Link to dev deployment in Amplify: 
https://display-hide-navbar.d1ms51d9oybzjy.amplifyapp.com/

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by briannakeune** [comment]
Would you mind

- [ ]  Filling out the PR template
- [ ] Resolving the merge conflicts
- [ ] Deploying a staged area for these changes

**Comment by BenSolt** [comment]
Finished it all! we are good to go. I did just realize that we need a navbar for the Login page that directs you to Register. Going to make that another feature, new trello card etc.

**Comment by moviedatascience** [comment]
This one has merge conflicts:


Conflicting files
src/App.js


Also can you fill out the template above, deleting the options that do not apply to this PR?

**Comment by moviedatascience** [comment]
If this is really a breaking change, what functionality does this break? Is someone currently looking to fix that functionality if this gets merged?

**Comment by BenSolt** [comment]
> If this is really a breaking change, what functionality does this break? Is someone currently looking to fix that functionality if this gets merged?

No it is not a breaking change, does nto break functionality. I checked that b/c i thought it was in relation to causing merge conflict issues. Sorry for the confusion.

**Comment by BenSolt** [comment]
> Having trouble with this locally, seems to hide navigation when a navLink item is clicked

Your right.. I did this incorrectly... I am really frustrated now..

**Comment by moviedatascience** [comment]
Added the dev deployment link, removed the bug reference, and commented in where the description of the feature should be placed.

File: package.json [hunk 1]
```diff
@@ -47,5 +47,6 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  }
+  },
+  "devDependencies": {}
```
Comment (briannakeune) [inline package.json]: Do we have devDependencies?

File: src/components/auth/_Register.scss [hunk 2]
```diff
@@ -10,10 +10,22 @@
     rgba(0, 0, 0, 0.85)
   );
 
-  display: flex;
-  flex-direction: row;
-  justify-content: space-evenly;
+  // display: flex;
```
Comment (briannakeune) [inline src/components/auth/_Register.scss]: are we using this?

File: src/components/auth/login.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ class LoginPage extends React.Component {
   handleChange = e => {
     const { name, value } = e.target;
     let errors = this.state.errors;
-    e.preventDefault();
+    // e.preventDefault();
```
Comment (briannakeune) [inline src/components/auth/login.js]: are we using this?

File: src/components/dashboard/navigation.js [hunk 4]
```diff
@@ -27,17 +27,19 @@ class Navigation extends Component {
...
       <div className="mainContainer" data-test={ifDev("navigation")}>
+        <div className='navContainer'>
         <div className="Bars">
           <FontAwesomeIcon className="bars-icon" icon={faBars} />
           <i className="far fa-bars"></i>
         </div>
 
         <div className="Links">
-          <NavLink className="Groa NavLink" to="/">
+          <NavLink className="Groa-NavLink" to="/">
```
Comment (briannakeune) [inline src/components/dashboard/navigation.js]: Why was this changed to Groa-NavLink?

File: src/App.test.js [hunk 5]
```diff
@@ -1,24 +1,44 @@
 import React from "react";
-import { render } from "@testing-library/react";
-import { getAllByTestId } from "./utils/test-utils.js";
+import '@testing-library/jest-dom/extend-expect'
+import { render, fireEvent } from "@testing-library/react";
+import { getByTestId, getAllByTestId } from "./utils/test-utils.js";
 
 // component to be tested.
 import App from "./App.js";
+import Login from "../src/components/auth/login";
```
Comment (briannakeune) [inline src/App.test.js]: Why was Login tested in this file?

Comment (cmruss) [reply src/App.test.js]: To test that the components dynamically render on App.js

File: src/components/dashboard/navigation.js (hunk 4 above)
Comment (cmruss) [reply src/components/dashboard/navigation.js]: Ben changed this for styling purposes to set the Home link apart, following an earlier spec of the design file, can be changed to whatever going forward

File: src/components/auth/login.js (hunk 3 above)
Comment (cmruss) [reply src/components/auth/login.js]: Removed

File: src/components/auth/_Register.scss (hunk 2 above)
Comment (cmruss) [reply src/components/auth/_Register.scss]: Removed

File: package.json (hunk 1 above)
Comment (cmruss) [reply package.json]: Going forward we will want all non-production dependencies on dev, so not yet but on future refactoring
